app/hooks/header.ts

```diff
@@ -17,53 +17,61 @@ type HeaderScrollContext = {
 
 export const MAX_OVERSCROLL = 80;
 
-export const useDefaultHeaderHeight = () => {
+export const useDefaultHeaderHeight = (hasSearch = false) => {
+    const insets = useSafeAreaInsets();
+    const isIOS = Platform.OS === 'ios';
     const isTablet = useIsTablet();
 
     if (isTablet) {
-        return ViewConstants.TABLET_HEADER_HEIGHT;
+        const tabletHeight = hasSearch ? ViewConstants.TABLET_SEARCH_HEIGHT_COLLAPSED : ViewConstants.TABLET_HEADER_HEIGHT;
+        return tabletHeight;
     }
 
-    if (Platform.OS === 'ios') {
-        return ViewConstants.IOS_DEFAULT_HEADER_HEIGHT;
+    let height = isIOS ? ViewConstants.IOS_DEFAULT_HEADER_HEIGHT : ViewConstants.ANDROID_DEFAULT_HEADER_HEIGHT;
+    if (hasSearch) {
+        height += isIOS ? 7 : 53;
```

Comment:
can you explain this values?

Reply:
I made some changes to help clear this code up.  

The new `hooks/header.ts` calculations for `defaultHeader` and `largeHeaderHeight` now have `if (hasSearch)` block that will return the values for the collapsed and expanded search headers heights. 

Looking in `constants/view.ts` the calculations are made in there and a single const is used for each height.

<img width="894" alt="image" src="https://user-images.githubusercontent.com/7575921/196839465-eb9ebb20-01e3-4bd8-bcb5-c3690e64d3e5.png">

`UNLOCKED_SEARCH_BOTTOM_MARGIN` and `LOCKED_SEARCH_BOTTOM_MARGIN` are the total margin calculations to get the search input located above the intial and results components (with 14x margin added per UX)

`TABLET_SEARCH_HEIGHT_COLLAPSED` and `MOBILE_SEARCH_HEIGHT_COLLAPSED` are the absolute heights for the collapsed states

`TABLET_SEARCH_MARGIN_EXPANDED` is the additional margin added to the expanded search header.
The total expanded search height is calculated in `useLargeHeaderHeight` by adding `defaultHeight + ViewConstants.LARGE_HEADER_TITLE_HEIGHT + TABLET_SEARCH_MARGIN_EXPANDED` where 
* defaultHeight - is based on the `SEARCH_INPUT_HEIGHT` + `14px margin` + `insets.top`
* `LARGE_HEADER_TITLE_HEIGHT` is the height of the large text height
* `TABLET_SEARCH_MARGIN_EXPANDED` is the additional margin request from UX